geomstats/learning/em_expectation_maximization.py

```diff
@@ -0,0 +1,265 @@
+        self.mixture_coefficients = gs.mean(posterior_probabilities, 0)
+
+    def update_means(self, data, posterior_probabilities,
+                     lr_means, tau_means, max_iter=DEFAULT_MAX_ITER):
+        """Means update function."""
+        n_gaussian = posterior_probabilities.shape[-1]
+
+        mean = FrechetMean(
+            metric=self.riemannian_metric,
+            method=self.mean_method,
+            lr=lr_means,
+            tau=tau_means,
+            max_iter=max_iter,
+            point_type=self.point_type)
+
+        data_expand = gs.expand_dims(data, 1)
+        data_expand = gs.repeat(data_expand, n_gaussian, axis=1)
+
+        mean.fit(data_expand, weights=posterior_probabilities)
+        self.means = gs.squeeze(mean.estimate_)
+
+    def update_variances(self, data, posterior_probabilities):
+        """Variances update function."""
+        n_data, n_gaussian = data.shape[0], self.means.shape[0]
+
+        data_expand = gs.expand_dims(data, 1)
+        data_expand = gs.repeat(data_expand, n_gaussian, axis=1)
+        means = gs.expand_dims(self.means, 0)
+        means = gs.repeat(means, n_data, axis=0)
+
+        weighted_dist_means_data = ((self.riemannian_metric.dist(
+            data_expand, means) ** 2) *
+            posterior_probabilities).sum(0) / \
+            posterior_probabilities.sum(0)
+
+        self.variances = \
+            self.normalization_factor.find_variance_from_index(
+                weighted_dist_means_data)
+
+    def _expectation(self, data):
+        """Update the posterior probabilities."""
+        probability_distribution_function = \
+            GaussianDistribution.gaussian_pdf(data,
+                                              self.means,
+                                              self.variances,
+                                              norm_func=self.
+                                              normalization_factor.
+                                              find_normalisation_factor,
+                                              metric=self.riemannian_metric)
+
+        if (probability_distribution_function.mean() !=
+                probability_distribution_function.mean()):
+            logging.info('EXPECTATION : Probability distribution function'
+                         'contain elements that are not numbers')
+
+        num_normalized_pdf = gs.einsum('j,...j->...j',
+                                       self.mixture_coefficients,
+                                       probability_distribution_function)
+        valid_pdf_condition = gs.amin(gs.sum(num_normalized_pdf, -1))
+
+        if valid_pdf_condition <= PDF_TOL:
+
+            num_normalized_pdf[gs.sum(num_normalized_pdf, -1) <= PDF_TOL] = 1
+
+        sum_pdf = gs.sum(num_normalized_pdf, -1)
+        posterior_probabilities =\
+            gs.einsum('...i,...->...i', num_normalized_pdf, 1 / sum_pdf)
+
+        if (gs.mean(posterior_probabilities) !=
```

Comment:
Is there a better way to express this condition? gs.any(mean_post is None) or something?